### PR TITLE
rocksdb: remove partial merge dead code

### PR DESCRIPTION
**Pull request by vivekmenezes**
This change is [Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/5502)

### CODE REVIEW

**Comment by vivekmenezes** [comment]
Looks like this might be called from rocksdb. I'm closing this PR for now.

**Comment by vivekmenezes** [comment]
Looks ok. Assigning to Peter Mattis for review.

File: storage/engine/rocksdb/db.cc [hunk 1]
```diff
@@ -751,27 +742,7 @@ class DBMergeOperator : public rocksdb::MergeOperator {
...
-    }
-
-    if (!meta.SerializeToString(new_value)) {
-      rocksdb::Warn(logger, "serialization error");
-      return false;
-    }
-    return true;
-  }
-
-  virtual bool PartialMergeMulti(
```
Comment (petermattis) [inline storage/engine/rocksdb/db.cc]: This code is called from RocksDB. It definitely isn't dead code that can just be removed.

Comment (vivekmenezes) [inline storage/engine/rocksdb/db.cc]: Can this code be called outside a raft command?

**Comment by mrtracy** [comment]
This code can definitely be called outside of a raft command.  We are not calling it, but RocksDB is. You can read all about it at https://github.com/facebook/rocksdb/wiki/Merge-Operator

(Note that "PartialMergeMulti" does not appear to be documented there, but it is available. It is a slightly optimized version of PartialMerge).

**Pull request closed without merging**